### PR TITLE
Don't trigger a key event if a key with the same associated char was pressed

### LINKED ISSUE #13770: Keypad input doubles as digit input
### Minetest version

```true
Minetest 5.7.0 (Windows)
Using Irrlicht 1.9.0mt10
Using LuaJIT 2.1.0-beta3
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="."
STATIC_LOCALEDIR="locale"
```

### Active renderer

_No response_

### Irrlicht device

_No response_

### Operating system and version

Windows 10

### CPU model

_No response_

### GPU model

_No response_

### OpenGL version

4.6.0

### Summary

I am left handed, when changing binds for WASD to Keypad: 5123, trying to move changes the active item to the corresponding number too

### Steps to reproduce

change the binding for W to Keypad5, try to move forward in a game, active item changes to 5 as well.

### PR DESCRIPTION
**Pull request by savilli**
Fix #13770
See https://github.com/minetest/minetest/issues/13770#issuecomment-1703328075

## To do

This PR is Ready for Review.

## How to test

Check that pressing a numpad key doesn't change the active item

### CODE REVIEW

**Review by grorp — APPROVED** [review]
Confirmed the bug on Linux with non-SDL Irrlicht. The fix works and makes sense.

The fix also results in being unable to change your viewing range using the numpad <kbd>+</kbd> and <kbd>-</kbd> keys with the default keymap, but I don't think that's a problem (this is expected, the non-numpad versions of the keys still work).

I can't reproduce the bug on Linux with SDL Irrlicht. If you press a numpad key there, `Char` always seems to be set to 0.

**Comment by sfan5** [comment]
Consider adding a short unit test for `operator==`, I believe we had some keycode test somewhere already.

**Comment by savilli** [comment]
> Consider adding a short unit test for `operator==`, I believe we had some keycode test somewhere already.

Done

File: src/unittest/test_keycode.cpp [hunk 1]
```diff
@@ -111,7 +111,7 @@ void TestKeycode::testCompare()
 {
 	// Basic comparison
 	UASSERT(KeyPress("5") == KeyPress("KEY_KEY_5"));
-	UASSERT(!(KeyPress("5") == KeyPress("KEY_NUMPAD_5")));
+	UASSERT(!(KeyPress("5") == KeyPress("KEY_NUMPAD5")));
```
Comment (savilli) [inline src/unittest/test_keycode.cpp:114]: This was broken for a while

**Pull request merged**